pkgs/checks/lib/src/checks.dart

```diff
@@ -261,7 +261,8 @@ abstract class Context<T> {
   /// context. The [label] will be used as if it were a single line "clause"
   /// passed to [expect]. If the label is empty, the clause will be omitted. The
   /// label should only be left empty if the value extraction cannot fail.
-  Subject<R> nest<R>(String label, Extracted<R> Function(T) extract,
+  Subject<R> nest<R>(
+      Iterable<String> Function() label, Extracted<R> Function(T) extract,
```

Comment:
Do you think we should consider taking this to the extreme of taking callbacks for `actual` and `which`?

https://github.com/dart-lang/test/pull/1901/files

I was thinking we shouldn't because it adds quite a bit of noise, but those two descriptions are also sometimes not read despite a value getting rejected, so it would be consistent to defer any formatting there too.